lib/stream_data.ex

```diff
@@ -823,37 +823,10 @@ defmodule StreamData do
   #       |> LazyTree.flatten()
   #     end)
   #
-  @spec list_of(t(a), keyword()) :: t([a]) when a: term()
-  def list_of(data, options) do
-    list_length_range_fun = list_length_range_fun(options)
-
-    new(fn seed, size ->
-      {min_length, max_length} = list_length_range_fun.(size)
-      {length, next_seed} = uniform_in_range(min_length, max_length, seed)
-
-      data
-      |> call_n_times(next_seed, size, length, [])
-      |> LazyTree.zip()
-      |> LazyTree.map(&list_lazy_tree(&1, min_length))
-      |> LazyTree.flatten()
-    end)
-  end
-
-  @doc """
-  Generates lists where each values is generated by the given `data`.
 
-  The same as calling `list_of/2` with `[]` as options.
-  """
-  def list_of(data) do
-    new(fn seed, size ->
-      {length, next_seed} = uniform_in_range(0, size, seed)
-
-      data
-      |> call_n_times(next_seed, size, length, [])
-      |> LazyTree.zip()
-      |> LazyTree.map(&list_lazy_tree(&1, 0))
-      |> LazyTree.flatten()
-    end)
+  @spec list_of(t(a), keyword()) :: t([a]) when a: term()
+  def list_of(data, options \\ []) do
+    unfold(nil, fn _ -> {data, nil} end, options)
```

Comment:
Please verify how this compares performance-wise to the previous implementation. `list_of` is used in many other combinators, so it heavily impacts overall performance of the library.

Reply:
Good point. Do you have a kind of testbed and performance data as bench mark from your performance optimisation activities? Would be nice to do it the same way.

Reply:
First results running 
`MIX_ENV=prod mix profile.fprof -e 'StreamData.list_of(StreamData.integer()) |> Enum.take(1000)'`

Test | Function | ACC(ms) | OWN(ms) | Percentage(acc)
-----|---------|-----------|-----------|---
`list_of` | `:fprof.apply_start_stop/4` | 1811.293  | 0.008 | 100.00 %
`list_of`  | anonymous `fn/3` in `StreamData.list_of/1` | 1783.694 | 8.326 | 100.00 %
`unfold`| `:fprof.apply_start_stop/4` | 2102.877 | 0.010 | 116.09 %
`unfold`|  anonymous `fn/4` in `StreamData.unfold/2` |  2073.250 | 15.602 | 116.23 %

The additional indirection of `unfold` seems to cost 16% more performance. Looks like I eat your performance optimisations. So `list_of` should not be implemented in terms of `unfold`. I will correct that.